common/Tests/Framework/src/AITestClass.ts

```diff
@@ -15,6 +15,15 @@ export interface FakeXMLHttpRequest extends XMLHttpRequest {
     respond: (status: number, headers: any, body: string) => void;
 }
 
+function _getObjName(target:any, unknownValue?:string) {
+    if (target.hasOwnProperty("prototype")) {
+        // Look like a prototype
+        return target.name || unknownValue || ""
+    }
+
+    return (((target || {})["constructor"]) || {}).name || unknownValue || "";
```

Comment:
Is this always the object type name? like "Object", "String", "Number", etc?

Reply:
Depending on the runtime environment, es3 doesn't have obj.constructor (which is why the ((target || {})["constructor"]) || {}).

I also pinched this from what I did in dynamicProto 😄